app/webhooks/stripe/route.ts

```diff
@@ -82,33 +83,20 @@ export async function POST(req: Request) {
 					);
 				}
 				await handleSubscriptionCancellation(event.data.object);
+				await upsertSubscription(event.data.object.id);
```

Comment:
:memo: Add upsertSubscription call after handleSubscriptionCancellation to ensure subscription data is properly synchronized in the database when a subscription is canceled.
This ensures our system maintains accurate subscription state after cancellation processing.